ui/page/fileListPublished/view.jsx

```diff
@@ -42,12 +58,58 @@ function FileListPublished(props: Props) {
     checkPendingPublishes();
   }, [checkPendingPublishes]);
 
+  const filteredClaims = useMemo(() => {
+    if (fetching) {
+      return [];
+    }
+
+    return myClaims.filter((claim) => {
+      const value = claim.value || {};
+      const src = value.source || {};
+      const title = (value.title || '').toLowerCase();
+      const description = (value.description || '').toLowerCase();
+      const tags = (value.tags || []).join('').toLowerCase();
+      const srcName = (src.name || '').toLowerCase();
+      const lowerCaseSearchText = searchText.toLowerCase();
+      const textMatches =
+        !searchText ||
+        title.indexOf(lowerCaseSearchText) !== -1 ||
+        description.indexOf(lowerCaseSearchText) !== -1 ||
+        tags.indexOf(lowerCaseSearchText) !== -1 ||
+        srcName.indexOf(lowerCaseSearchText) !== -1;
+      return textMatches && filterBy.includes(claim.value_type);
+    });
+  }, [fetching, myClaims, filterBy, searchText]);
+
+  const urlTotal = filteredClaims.length;
+
+  const urls = useMemo(() => {
+    const params = JSON.parse(paramsString);
+    const zeroIndexPage = Math.max(0, params.page - 1);
+    const paginated = filteredClaims.slice(
+      zeroIndexPage * params.page_size,
+      zeroIndexPage * params.page_size + params.page_size
+    );
+    return paginated.map((claim) => claim.permanent_url);
+  }, [filteredClaims, paramsString]);
+
+  const { search } = location;
+
+  // Go back to the first page when the filtered claims change.
+  // This way, we avoid hiding results just because the
+  // user may be on a different page (page that was calculated
+  // using a different state, ie, different filtered claims)
   useEffect(() => {
-    if (paramsString && fetchClaimListMine) {
-      const params = JSON.parse(paramsString);
-      fetchClaimListMine(params.page, params.page_size, true, filterBy.split(','));
+    if (!disableHistory) {
+      const params = new URLSearchParams(search);
+      params.set(PAGINATE_PARAM, '1');
+      history.push('?' + params.toString());
     }
-  }, [paramsString, filterBy, fetchClaimListMine]);
+  }, [filteredClaims]);
```

Comment:
@jessopb I'm not entirely happy with this but I couldn't think of a better way to work around this issue.

Reply:
Seems to work really well. I don't think you need "disableHistory" - that mostly doesn't exist except if it's passed to <Paginate as a prop, and only on one page.

Reply:
Updated. Thanks @jessopb for the comment.